sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ToPrettyString.scala

```diff
@@ -51,7 +52,13 @@ case class ToPrettyString(child: Expression, timeZoneId: Option[String] = None)
 
   override protected val binaryFormatter: BinaryFormatter = ToStringBase.getBinaryFormatter
 
-  private[this] lazy val castFunc: Any => UTF8String = castToString(child.dataType)
+  private[this] lazy val castFunc: Any => UTF8String = child.dataType match {
+    case CharType(length) if SQLConf.get.preserveCharVarcharTypeInfo =>
+      castToChar(child.dataType, length)
```

Comment:
This is wrong. `ToPrettyString` always cast the child data type to plain string type. So the cast target is always plain string.